src/aliceVision/lensCorrectionProfile/lcp.hpp

```diff
@@ -0,0 +1,733 @@
+#pragma once
+
+#include <boost/filesystem/path.hpp>
+
+#include <string>
+#include <vector>
+#include <sstream>
+#include <map>
+
+enum class LCPCorrectionMode
+{
+    VIGNETTE,
+    DISTORTION,
+    CA
+};
+
+enum class LCPReadingState
+{
+    WaitSequence,
+    FillCommonAndCameraSettings,
+    FillCameraSettings,
+    FillGeometryModel,
+    FillVignetteModel,
+    FillChromaticGreenModel,
+    FillChromaticBlueGreenModel,
+    FillChromaticRedGreenModel,
+    WaitNewModel
+};
+
+inline std::ostream& operator<<(std::ostream& os, const LCPReadingState& state)
+{
+    if (state == LCPReadingState::WaitSequence)
+        os << "WaitSequence";
+    else if (state == LCPReadingState::FillCommonAndCameraSettings)
+        os << "FillCommonAndCameraSettings";
+    else if (state == LCPReadingState::FillCameraSettings)
+        os << "FillCameraSettings";
+    else if (state == LCPReadingState::FillGeometryModel)
+        os << "FillGeometryModel";
+    else if (state == LCPReadingState::FillVignetteModel)
+        os << "FillVignetteModel";
+    else if (state == LCPReadingState::FillChromaticGreenModel)
+        os << "FillChromaticGreenModel";
+    else if (state == LCPReadingState::FillChromaticBlueGreenModel)
+        os << "FillChromaticBlueGreenModel";
+    else if (state == LCPReadingState::FillChromaticRedGreenModel)
+        os << "FillChromaticRedGreenModel";
+    else if (state == LCPReadingState::WaitNewModel)
+        os << "WaitNewModel";
+
+    return os;
+}
+
+
+struct settingsInfo
+{
+    float FocalLength = 0.f;
+    float FocusDistance = 0.f;
+    float ApertureValue = 0.f;
+
+    void reset()
+    {
+        FocalLength = 0.f;
+        FocusDistance = 0.f;
+        ApertureValue = 0.f;
+    }
+};
+
+struct RectilinearModel
```

Comment:
These `Model` structs should be documented so we can understand what the parameters represent and how they relate to the other classes we have in the project (for example I'm guessing this one is related to `camera::Distortion`)

Reply:
Done